rest-api/test/unit_test/api_test/biobank_order_api_test.py

```diff
@@ -17,9 +19,205 @@ def setUp(self):
+    result = self.send_post(self.path, order_json)
+    load_biobank_order_json(self.participant.participantId,
+                            filename='biobank_order_1.json')
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all better",
+      "amendedInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      }
+    }
+    get_order = self.send_get(path)
+    full_order = get_order.copy()
+    full_order.update(request_data)
+    self.send_put(path, request_data=full_order, headers={'If-Match': 'W/"1"'})
+
+    get_amended_order = self.send_get(path)
+    self.assertEqual(get_amended_order['version'], 2)
+    self.assertEqual(get_amended_order['meta'], {'versionId': 'W/"2"'})
+    self.assertEqual(get_amended_order['amendedReason'], 'Its all better')
+    self.assertEqual(get_amended_order['amendedUsername'], 'fred@pmi-ops.org')
+    self.assertEqual(get_amended_order['amendedSiteId'], 1)
+
+  def test_amend_a_restored_order(self):
+    self.summary_dao.insert(self.participant_summary(self.participant))
+    order_json = load_biobank_order_json(self.participant.participantId,
+                                         filename='biobank_order_2.json')
+    result = self.send_post(self.path, order_json)
+    full_order_json = load_biobank_order_json(self.participant.participantId,
+                                              filename='biobank_order_1.json')
+    _strip_fields(result)
+    _strip_fields(full_order_json)
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all wrong",
+      "cancelledInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "cancelled"
+    }
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"1"'})
+    self.send_get(path)
+    request_data = {
+      "amendedReason": "I didnt mean to cancel",
+      "restoredInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "restored"
+    }
+
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"2"'})
+
+    request_data = {
+      "amendedReason": "Its all better",
+      "amendedInfo": {
```

Comment:
Mutate the samples and identifiers here, too, and verify they come back correctly below.

Reply:
done